builder/api/src/main/java/io/helidon/builder/api/Prototype.java

```diff
@@ -112,6 +113,29 @@ public interface ConfiguredBuilder<BUILDER, PROTOTYPE> extends Builder<BUILDER,
                          boolean allFromServiceLoader) {
             return ProvidedUtil.discoverServices(config, serviceLoader, providerType, configType, allFromServiceLoader);
         }
+
+        /**
+         * Discover service from configuration.
+         *
+         * @param config               configuration located at the node of the service providers
+         *                             (either a list node, or object, where each child is one service)
+         * @param serviceLoader        helidon service loader for the expected type
```

Comment:
does the service loader really need to be passed, or can it just be created directly?  When will we ever have a special loader?

Reply:
You cannot create it in the other module, due to module system. The module that declares `uses` is the one that must create the `ServiceLoader` instance (or we would have to do some complex stuff using reflection, which we do not do by design).